packages/amplify-graphql-auth-transformer/src/resolvers/mutation.update.ts

```diff
@@ -195,18 +195,31 @@ const dynamicGroupRoleExpression = (roles: Array<RoleDefinition>, fields: Readon
               methodCall(ref('util.defaultIfNull'), ref(`ctx.result.${role.entity!}`), entityIsList ? list([]) : nul()),
             ),
             set(ref(`ownerClaim${idx}`), getOwnerClaim(role.claim!)),
-            set(ref(`ownerAllowedFields${idx}`), raw(JSON.stringify(role.allowedFields))),
-            set(ref(`ownerNullAllowedFields${idx}`), raw(JSON.stringify(role.nullAllowedFields))),
+            role.allowedFields && set(ref(`ownerAllowedFields${idx}`), raw(JSON.stringify(role.allowedFields))),
```

Comment:
nit might also be good to trim the falsy values out at the end
```js
ownerExpression.filter(Boolean)
```

Reply:
updated